pkg/controller/templateupdaterequest/templateupdaterequest_controller.go

```diff
@@ -0,0 +1,272 @@
+package templateupdaterequest
+
+import (
+	"context"
+	"strings"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/codeready-toolchain/host-operator/pkg/controller/nstemplatetier"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_templateupdaterequest")
+
+// Add creates a new TemplateUpdateRequest Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, _ *configuration.Config) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileTemplateUpdateRequest{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("templateupdaterequest-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource TemplateUpdateRequest
+	err = c.Watch(&source.Kind{Type: &toolchainv1alpha1.TemplateUpdateRequest{}}, &handler.EnqueueRequestForObject{})
```

Comment:
I guess that we can ignore changes in the status part of the TemplateUpdateRequests - please add 
`, predicate.GenerationChangedPredicate{}`
I don't use suggestion because you would need to add the import manually anyway.

Reply:
ok, thanks ;) let me do that

Reply:
done in c73c3ff

Reply:
ah no we can't do that, otherwise there will be no reconcile loop when a TemplateUpdateRequest is marked as complete (with success or failure), and thus none will be deleted and the whole process will be blocked 😬

Reply:
reverted in 52c1e12

Reply:
ah never mind, I thought you were referring to the `NSTemplateTier` controller 🤦‍♂️ . I'll re-revert the change. In other words, I'll apply it for good this time 😬

Reply:
done in c9700ce